pkg/operator/ceph/cluster/cluster.go

```diff
@@ -157,9 +151,14 @@ func (c *cluster) createInstance(rookImage string) error {
 		return fmt.Errorf("failed to create override configmap %s. %+v", c.Namespace, err)
 	}
 
+	verImage := version.VersionedImage{
```

Comment:
Instead of this struct, I believe what we want is for the ceph spec to be passed around, and the version will just need to be a separate param, similar to the `rookImage`. I have a change in #2780 where at least the mons will follow the pattern of passing around the ceph spec, which will reduce the number of parameters to these methods. We should make the same change to other daemons as well to clean things up. For now, adding a new param for CephVersion will follow that new pattern. But feel free to comment on #2780 if you see a different pattern working better.

Reply:
When you say `ceph spec` do you mean `ClusterSpec`? If we end up passing around a pointer to the full spec, then I think your suggestion of having an additional parameter for the version would be just great. But as it stands right now, adding an additional parameter to pass around the detected version in addition to the version name (taken from the spec) is adding to the already criminal width of the cluster factory method signatures.

What's the timeline on #2780? I'm happy to refactor this on top of it if it is going to merge soonish.

Reply:
Correct, `ClusterSpec`. Tomorrow I'm hoping we can merge #2693, then I'll fix merge conflicts and merge #2780, then we could get this one.

Reply:
That sounds good. This one is a particularly rough rebase, so I'd like to do it only once more.